v2/pkg/engine/datasource/grpc_datasource/execution_plan_visitor.go

```diff
@@ -542,6 +559,69 @@ func (r *rpcPlanVisitor) buildMessageField(fieldName string, typeRef, parentType
 	r.planInfo.requestMessageAncestors = r.planInfo.requestMessageAncestors[:len(r.planInfo.requestMessageAncestors)-1]
 }
 
+func (r *rpcPlanVisitor) buildInputMessageField(typeRef int, fieldName, jsonPath string, dt DataType) RPCField {
+	field := RPCField{
+		Name:     fieldName,
+		Optional: !r.definition.TypeIsNonNull(typeRef),
+		TypeName: dt.String(),
+		JSONPath: jsonPath,
+	}
+
+	if r.definition.TypeIsList(typeRef) {
+		switch {
+		// for nullable or nested lists we need to build a wrapper message
+		// Nullability is handled by the datasource during the execution.
+		case r.typeIsNullableOrNestedList(typeRef):
+			field.ListMetadata = r.createListMetadata(typeRef)
+			field.IsListType = true
+		default:
+			// For non-nullable single lists we can directly use the repeated syntax in protobuf.
+			field.Repeated = true
+		}
+	}
+
+	return field
+}
+
+func (r *rpcPlanVisitor) typeIsNonNullList(typeRef int) bool {
```

Comment:
we coud add this helper to ast

and check smth like this type.Kind non null and typeOfType.kind == list